### ISSUE TITLE
How to change icon of a database row?

### ISSUE THREAD

**Issue by nirmitdesai** [jamalex/notion-py#154]
I opened a "name" column of an inline database as a page block. I want to upload local file as an icon for that page. Is it possible?

**Comment by Linpean** [comment]
I find a way  to resolve this problem. you should delete the those database row after you change the icon, then cancel deleting action. those row will show your icons.

**Comment by Linpean** [comment]
![批量设置ICON](https://user-images.githubusercontent.com/20737704/85806609-1558e400-b782-11ea-8910-cb39d7110796.gif)
like this

**Comment by nirmitdesai** [comment]
Hi LinPean,

I am still confused. Can you share your source code?  
Also, what do you mean by cancelling the delete action. From what I can see, you delete the rows but they got populated again?  
What's reason for deleting the rows?

Btw here's same gif slowed down : https://im5.ezgif.com/tmp/ezgif-5-86993a9b5b5b.gif

**Comment by nirmitdesai** [comment]
Also, I would suggest you delete the gif because someone could grab your Notion token. You can upload new gif and hide the token part.

**Comment by Linpean** [comment]
> 另外，我建议您删除gif，因为有人可以抓住您的Notion令牌。您可以上传新的gif并隐藏令牌部分。

The source code is not point. That CANCELLING THE DELETE is point.
At first, the code will upload images,  then you should delete that rows that have not show the icon normally, and cancelling deleting(I mean ctrl + z), the icon will show, that is magic , I also understand the reason,but it works.

this is source code to upload the image icon
```
# codeing:UTF-8
import mimetypes
import os
import random
import requests


from notion.client import NotionClient
from notion.block import CollectionViewBlock,ImageBlock
from notion.operations import build_operation


# 获得Token验证
client = NotionClient(token_v2="<tokenV2>")


# 输入附件库页面视图链接
cv = client.get_collection_view("https://www.notion.so/7c33a4905d2544f1a8af068eacfd55d1?v=980c47be43dc459fafc6c8e055f50e5e")

path="D:\\OneDrive\\02Design设计区\\20200306古剑ol_wiki\\古剑奇谭网络版技能资料\\斩风技能和奇穴图标篇\\新建文件夹" #设置需要上传的文件夹位置


# 遍历文件下所有文件
filelist=os.listdir(path)#该文件夹下所有的文件（包括文件夹）
i = 0
for files in filelist:#遍历所有文件
    i=i+1
    filedir=os.path.join(path,files)#原来的文件路径                
    if os.path.isdir(filedir):#如果是文件夹则跳过
            continue;
    filename=os.path.splitext(files)[0]#文件名（不包含路径）
    filetype=os.path.splitext(files)[1]#文件扩展名
    print(filedir, filename, filetype)
    


    #----开始进行Notion操作----
    for row in cv.collection.get_rows():
        if  row.title == filename:
            img_block = row.children.add_new(ImageBlock)
            img_block.upload_file(filedir)
            row.icon = img_block.source
print("完成Notion操作，退出程序")
exit()
```

**Comment by nirmitdesai** [comment]
Oh interesting. When do you have to delete and ctrl+z the rows? After the script has finished running?  
  
Thanks btw!

**Comment by Linpean** [comment]
> Oh interesting. When do you have to delete and ctrl+z the rows? After the script has finished running?
> 
> Thanks btw!

Yeah , after the script has finished, I deleted the rows and ctrl+z.
I guess that the deletint action  may be change url.
Anyway, it works.

**Comment by nirmitdesai** [comment]
Will give it a try. Good work!
